### DISCUSSION TITLE
How to use nested placeholder in localizationformatter ?

### DISCUSSION THREAD

**Discussion by fm119** [axuno/SmartFormat#339] [Q&A] [answered]
`Smart.Format("Account Status: {:L(en):{status}}", new Dictionary<string, string>(){ { "status", "Active" } })`

I expect the formatter to first replace {status} with the corresponding text from the dictionary, and then translate the resulting text into another language.

TemplateFormatter is not an option for me because there may be many cases for status.

**Comment by axunonb** [comment, marked as answer]
How to use nested placeholders in LocalizationFormatter is described [here in the Wiki](https://github.com/axuno/SmartFormat/wiki/Localization-_-LocalizationFormatter). In effect it works in the same way as [Localization in .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/localization). So you can add placeholders for variables, but the variables' values don't become a part of the text to translate. Localization is for literals.

**Reply by fm119** [reply]
I get it, thanks for your response

**Comment by zacateras** [comment]
@axunonb can functionality of parameterized localization as described in the example presented by @fm119 can be supported after extending LocalizationFormatter; will you accept a pr with such an extension?

**Reply by axunonb** [reply]
I'll have a look later this week.

**Reply by zacateras** [reply]
Draft implementation proposed as PR here: https://github.com/axuno/SmartFormat/pull/350

**Reply by zacateras** [reply]
@axunonb could you plan a release for #350 so that I can use this feature in my solutions? 😉

**Comment by axunonb** [comment]
Sure. There was another PR pending (support for `IReadOnlyDictionary`) which will go into the same version. Thanks for your patience. Release will be published later today or tomorrow.
